ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerSubscription.scala

```diff
@@ -141,9 +136,54 @@ object ParallelIndexerSubscription {
           logger.info(s"Storing ${update.description}")
       }
     }
-    val batch = input.iterator.flatMap { case ((offset, update), _) =>
+
+    val nonMetering = input.iterator.flatMap { case ((offset, update), _) =>
       toDbDto(offset)(update)
     }.toVector
+
+    val metering = input.iterator
+      .collect({
+        case (
+              (
+                offset,
+                TransactionAccepted(
+                  Some(CompletionInfo(_, applicationId, _, _, _, Some(statistics))),
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                  _,
+                ),
```

Comment:
I think unapplying here is more verbose (lines-wise) than just type ascription+dereferencing whichever fields you need

Reply:
Broken it into two stages to reduce line count.